centreon/src/Core/Security/Authentication/Infrastructure/Provider/SAML.php

```diff
@@ -0,0 +1,500 @@
+
+use Assert\AssertionFailedException;
+use Centreon;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Contact\Interfaces\ContactRepositoryInterface;
+use Centreon\Domain\Entity\ContactGroup;
+use Centreon\Domain\Log\LoggerTrait;
+use CentreonSession;
+use Core\Application\Configuration\User\Repository\WriteUserRepositoryInterface;
+use Core\Domain\Configuration\User\Model\NewUser;
+use Core\Security\AccessGroup\Domain\Model\AccessGroup;
+use Core\Security\Authentication\Application\Provider\ProviderAuthenticationInterface;
+use Core\Security\Authentication\Application\UseCase\Login\LoginRequest;
+use Core\Security\Authentication\Domain\Exception\AclConditionsException;
+use Core\Security\Authentication\Domain\Exception\AuthenticationConditionsException;
+use Core\Security\Authentication\Domain\Exception\SSOAuthenticationException;
+use Core\Security\Authentication\Domain\Model\AuthenticationTokens;
+use Core\Security\Authentication\Domain\Model\NewProviderToken;
+use Core\Security\Authentication\Infrastructure\Provider\Exception\InvalidArgumentProvidedException;
+use Core\Security\Authentication\Infrastructure\Provider\Exception\InvalidUserIdAttributeException;
+use Core\Security\Authentication\Infrastructure\Provider\Exception\SAML\InvalidMetadataException;
+use Core\Security\Authentication\Infrastructure\Provider\Exception\SAML\ProcessAuthenticationResponseException;
+use Core\Security\Authentication\Infrastructure\Provider\Exception\UserNotAuthenticatedException;
+use Core\Security\Authentication\Infrastructure\Provider\Settings\Formatter\SettingsFormatterInterface;
+use Core\Security\ProviderConfiguration\Domain\LoginLoggerInterface;
+use Core\Security\ProviderConfiguration\Domain\Model\Configuration;
+use Core\Security\ProviderConfiguration\Domain\Model\Provider;
+use Core\Security\ProviderConfiguration\Domain\SAML\Model\CustomConfiguration;
+use Core\Security\ProviderConfiguration\Domain\SecurityAccess\Conditions;
+use Core\Security\ProviderConfiguration\Domain\SecurityAccess\GroupsMapping as GroupsMappingSecurityAccess;
+use Core\Security\ProviderConfiguration\Domain\SecurityAccess\RolesMapping;
+use DateInterval;
+use DateTimeImmutable;
+use Exception;
+use OneLogin\Saml2\Auth;
+use OneLogin\Saml2\Error;
+use OneLogin\Saml2\Utils;
+use OneLogin\Saml2\ValidationError;
+use Pimple\Container;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Throwable;
+
+class SAML implements ProviderAuthenticationInterface
+{
+    use LoggerTrait;
+
+    /**
+     * @var Configuration
+     */
+    private Configuration $configuration;
+
+    /**
+     * @var Centreon
+     */
+    private Centreon $legacySession;
+
+    /**
+     * @var string
+     */
+    private string $username;
+
+    /**
+     * @var ContactInterface|null
+     */
+    private ?ContactInterface $authenticatedUser = null;
+
+    /**
+     * @var Auth|null
+     */
+    private ?Auth $auth = null;
```

Comment:
php-cs-fixer missing
```suggestion
    /** @var Configuration */
    private Configuration $configuration;

    /** @var Centreon */
    private Centreon $legacySession;

    /** @var string */
    private string $username;

    /** @var ContactInterface|null */
    private ?ContactInterface $authenticatedUser = null;

    /** @var Auth|null */
    private ?Auth $auth = null;
```